components/suggest/src/db.rs

```diff
@@ -867,6 +867,23 @@ impl<'a> SuggestDao<'a> {
         )
     }
 
+    pub fn is_exposure_suggestion_type_ingested(&self, suggestion_type: &str) -> Result<bool> {
+        let type_exists = self.conn.exists(
+            r#"
+            SELECT
+              type
+            FROM
+              exposure_custom_details
+            WHERE
+              type = :type
+            "#,
```

Comment:
I'm not sure this will handle all the corner cases.  The main one is that if an exposure type is not enabled, I think this will always be true and we'll always try to process the record.  Maybe that's not a big deal though, since we'll quickly skip over it in `ingest_record()`.  I worry that there could be other, more tricky corner cases though that end up affecting the user.  For example:
- Fx version 150 ingests `exposure-record-foo-100`
- Fx version 151 disables `foo` and therefore skips `exposure-record-foo-200`
- Fx version 152 re-enables `foo`, but then doesn't ingest `exposure-record-foo-200` since it's unchanged and we have an suggestions from `exposure-record-foo-100`.

I'm not sure how realistic that one is, but I'm a bit worried that something like that might happen.  What if you added a field to `ingest_records` that tracks if it matched the constraints when we last processed it?  If that's false but the record now does match the constraints, then we process it as a new record.

Reply:
> The main one is that if an exposure type is not enabled, I think this will always be true and we'll always try to process the record. Maybe that's not a big deal though, since we'll quickly skip over it in `ingest_record()`.

Right, that's my opinion since otherwise `should_reingest_record()` would need to duplicate the logic inside `ingest_record()` that determines whether the attachment should be ingested. It would need to take on part of the role that `ingest_record()` already does.

I think the names of these methods might be misleading in this context. I almost want to say `process_record()` or `handle_record()` instead of `ingest_record()`. Ingest encompasses processing, downloading, and updating the DB. If `should_reingest_record()` returns true, it doesn't mean all those things need to happen, just that the record should be re-processed and then maybe they should happen.

> I'm not sure how realistic that one is, but I'm a bit worried that something like that might happen. What if you added a field to `ingest_records` that tracks if it matched the constraints when we last processed it?

I had a similar thought but was hoping we could get away without something like that. :-) My thought was, we could avoid those problems either by updating the record on RS or by using a new exposure suggestion type altogether. That would require some manual work on our part, which we might forget to do. I think for the use case we have in mind for exposure suggestions -- running one-off experiments -- we're fine, but you're probably right it should be addressed in the code, so I'll take look.

Reply:
This turned out to be pretty simple to fix: If there's no row in `suggestions` with the `record_id` of the unchanged record, then its attachment was not ingested, and so the record should be re-processed. (And then if the ingest provider constraints match the record's exposure type, `ingest_record()` should go on to ingest its attachment.) I don't think I'm overlooking anything?

Reply:
I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1915715